Remove `push` workflow for jobs that already run on PR and merge

jentfoo opened the pull request:
An attempt to reduce our Actions usage this PR removes the workflow execution for `push` actions on several jobs.

jentfoo commented on .github/workflows/unit-tests-code.yaml:
```diff
@@ -2,10 +2,6 @@ name: Unit Tests (Go)
 run-name: Unit Tests (Go) - ${{ github.run_id }} - @${{ github.actor }}
 
 on:
-  push:
-    branches:
-      - master
-      - branch/*
```
Should I revert all unit tests or are there only specific jobs doing this monitoring?

r0mant replied:
We should keep it for Go unit tests and integration tests (both root and non-root).

jentfoo replied:
Thank you @r0mant 

I reverted this change for the 4 files now referenced in the commit message here: https://github.com/gravitational/teleport/commit/3937b98dda5f0479495ed841779abf433e1a8a8b

Let me know if there are any other gotchas you see here

jentfoo committed:
Remove `push` workflow for jobs that already run on PR and merge

In an attempt to reduce our Actions usage this PR removes the workflow execution for `push` actions on several jobs.

The following files were left as an exception to make sure flakey tests are discovered:
* integration-tests-non-root.yaml
* integration-tests-root.yaml
* unit-tests-code.yaml
* unit-tests-integrations.yaml
